problem/templates/problem/editTab/tag.html

```diff
@@ -34,12 +42,39 @@
   <tr>
     <th>Add Tag</th>
     <td>
-      <div class='input-group'>
-        <input type='text' id='newTag' class='form-control' width='25%'>
-        <span class='input-group-btn'>
-          <button id='add_tag_button' class='btn btn-default'>Add</button>
-        </span>
-      </div>
+      <form id="addTag">
+        <div class='input-group'>
+          {% csrf_token %}
+          <input type='text' id='newTag' name='tag' class='form-control' width='25%'>
+          <span class='input-group-btn'>
+            <button id='add_tag_button' class='btn btn-default'>Add</button>
+          </span>
+        </div>
+      </form>
     </td>
   </tr>
 </table>
+<script>
+$(document).ready(function() {
+    $("#addTag").submit(function(e) {
+        e.preventDefault();
+        return add_new_tag({{pid}});
+    });
+    $(".del_tag_btn").click(function (e) {
+        var row = $(this).parents("tr");
+        var tid = $(this).parents("tr").attr('data-target');
+        $.ajax({
+            type: 'GET',
+            url: '/problem/'+{{pid}}+'/tag/'+tid+'/delete/',
+            success: function(data) {
+              row.hide();
+            }
+        });
+        return false;
+    });
+});
+</script>
+{% else %}
+<strong>Save your problem before editing the tags</strong>
+{% endif %}
+
```

Comment:
For overall of this file your ' and " usage may be same.
e.g. 
\<div class='panel panel-default' style='float:right;width:47%'>
and
\<div class="tab-pane" id="tag">
Their quotation usages are different.